Lab/demo/Lab/Plugins/IO/OM_io_plugin.cpp

```diff
@@ -0,0 +1,175 @@
+  bool save(QFileInfo fileinfo,QList<CGAL::Three::Scene_item*>&);
+};
+
+QString CGAL_Lab_om_plugin::nameFilters() const {
+  return "om files (*.om)";
+}
+
+bool CGAL_Lab_om_plugin::canLoad(QFileInfo) const {
+  return true;
+}
+
+
+
+QList<Scene_item*>
+CGAL_Lab_om_plugin::
+load(QFileInfo fileinfo, bool& ok, bool add_to_scene){
+
+  // Open file
+  std::ifstream in(fileinfo.filePath().toUtf8(), std::ios::in | std::ios::binary);
+  if(!in) {
+    std::cerr << "Error! Cannot open file " << (const char*)fileinfo.filePath().toUtf8() << std::endl;
+    ok = false;
+    return QList<Scene_item*>();
+  }
+  in.close();
+  if(fileinfo.size() == 0)
+  {
+    CGAL::Three::Three::warning( tr("The file you are trying to load is empty."));
+    Scene_surface_mesh_item* item = new Scene_surface_mesh_item();
+    item->setName(fileinfo.completeBaseName());
+    ok = true;
+    if(add_to_scene)
+      CGAL::Three::Three::scene()->addItem(item);
+    return QList<Scene_item*>()<<item;
+  }
+
+  try
+  {
+    typedef boost::graph_traits<SMesh>::vertex_descriptor vertex_descriptor;
+    typedef boost::graph_traits<SMesh>::edge_descriptor edge_descriptor;
+    std::map<vertex_descriptor,bool> sm_vfeature_map;
+    auto sm_vfeature_pmap = boost::make_assoc_property_map(sm_vfeature_map);
+
+    std::map<edge_descriptor,bool> sm_efeature_map;
+    auto sm_efeature_pmap = boost::make_assoc_property_map(sm_efeature_map);
+
+    // Try building a surface_mesh
+    SMesh* SM = new SMesh();
+    ok = CGAL::IO::read_OM((const char*)fileinfo.filePath().toUtf8(), *SM, sm_vfeature_pmap, sm_efeature_pmap);
+
+    if(!ok || !SM->is_valid() || SM->is_empty()){
+      std::cerr << "Error: Invalid facegraph" << std::endl;
+    }
+    else{
+      Scene_surface_mesh_item* item = new Scene_surface_mesh_item(SM);
+
+      Scene_polyhedron_selection_item* selection_item = new Scene_polyhedron_selection_item(item, CGAL::Three::Three::mainWindow());
+      bool eselected = false;
+      bool vselected = false;
+      for(auto v : vertices(*SM)){
+        if(get(sm_vfeature_pmap, v)){
+          selection_item->selected_vertices.insert(v);
+          vselected = true;
+        }
+      }
+      for(auto e : edges(*SM)){
+        if(get(sm_efeature_pmap, e)){
+          selection_item->selected_edges.insert(e);
+          eselected = true;
+        }
+      }
+      item->setName(fileinfo.completeBaseName());
+      ok = true;
+      if(add_to_scene){
+        CGAL::Three::Three::scene()->addItem(item);
+        CGAL::Three::Three::scene()->addItem(selection_item);
+      }
+      return QList<Scene_item*>()<<item << selection_item;
+    }
+  }
+  catch(...){}
+
+  ok = false;
+  return QList<Scene_item*>();
+}
+
+bool CGAL_Lab_om_plugin::canSave(const CGAL::Three::Scene_item* item)
+{
+  return qobject_cast<const Scene_surface_mesh_item*>(item);
+}
+
+bool CGAL_Lab_om_plugin::
+save(QFileInfo fileinfo,QList<CGAL::Three::Scene_item*>& items)
+{
+#if 0
```

Comment:
#if 0

Reply:
Saving a mesh in  `*.om` format is  not yet done.  It would be cleaner to reply `false`  in a `can_save()` method.